src/components/Explore/ShareButton.tsx

```diff
@@ -0,0 +1,139 @@
+const TWITTER_HEIGHT = 480
+const SHADOW = '0px 1px 6px rgba(0, 0, 0, 0.9), 0px 8px 12px rgba(13, 14, 14, 0.8)'
+
+const ShareButtonDisplay = styled.div`
+  display: flex;
+  cursor: pointer;
+  position: relative;
+
+  &:hover {
+    color: ${({ theme }) => darken(0.1, theme.text2)};
+  }
+`
+const ShareActions = styled.div`
+  position: absolute;
+  top: 28px;
+  right: 0px;
+  padding: 8px 0px;
+  display: flex;
+  flex-direction: column;
+  width: fit-content;
+  overflow: auto;
+  background-color: ${({ theme }) => theme.bg0};
+  border: 1px solid ${({ theme }) => theme.bg1};
+  box-shadow: ${SHADOW};
+  border-radius: 12px;
+`
+const ShareAction = styled.div`
+  display: flex;
+  align-items: center;
+  padding: 12px 16px;
+  gap: 8px;
+  width: 200px;
+  height: 48px;
+  color: ${({ theme }) => theme.text1};
+  cursor: pointer;
+
+  &:hover {
+    color: ${({ theme }) => darken(0.08, theme.text1)};
+    background-color: ${({ theme }) => darken(0.08, theme.bg1)};
+  }
+`
+const LinkCopied = styled.div<{ show: boolean }>`
+  display: ${({ show }) => (show ? 'flex' : 'none')};
+  width: 328px;
+  height: 72px;
+  color: ${({ theme }) => theme.text1};
+  background-color: ${({ theme }) => theme.bg0};
+  justify-content: flex-start;
+  align-items: center;
+  padding: 24px 16px;
+  position: absolute;
+  right: 32px;
+  bottom: 32px;
+  font-size: 14px;
+  gap: 8px;
+  border: 1px solid rgba(153, 161, 189, 0.08);
+  box-shadow: ${SHADOW};
+  border-radius: 20px;
+  animation: cssAnimation 0s ease-in 3s forwards;
+
+  @keyframes cssAnimation {
+    to {
+      width: 0;
+      height: 0;
+      overflow: hidden;
+      display: none;
+    }
+  }
+`
+export interface tokenInfo {
+  tokenName: string
+  tokenSymbol: string
+}
+
+export default function ShareButton(tokenInfo: { tokenName: string; tokenSymbol: string }) {
+  const theme = useTheme()
+  const node = useRef<HTMLDivElement | null>(null)
+  const open = useModalIsOpen(ApplicationModal.SHARE)
+  const toggleShare = useToggleModal(ApplicationModal.SHARE)
+  useOnClickOutside(node, open ? toggleShare : undefined)
+  const [showCopied, setShowCopied] = useState(false)
+  const positionX = (window.screen.width - TWITTER_WIDTH) / 2
+  const positionY = (window.screen.height - TWITTER_HEIGHT) / 2
+
+  const shareTweet = () => {
+    toggleShare()
+    window.open(
+      `https://twitter.com/intent/tweet?text=Check%20out%20${tokenInfo.tokenName}%20(${tokenInfo.tokenSymbol})%20https://app.uniswap.org/%23/tokens/${tokenInfo.tokenSymbol}%20via%20@uniswap`,
+      'newwindow',
+      `left=${positionX}, top=${positionY}, width=${TWITTER_WIDTH}, height=${TWITTER_HEIGHT}`
+    )
+  }
+  const copyLink = () => {
+    navigator.clipboard.writeText(window.location.href).then(
+      function () {
+        setShowCopied(true)
+        toggleShare()
+        setTimeout(() => setShowCopied(false), 3000)
+      },
+      function () {
+        console.log('Clipboard copy failed.')
```

Comment:
I think you should write explicit `useCallback` handlers for the success/error case here, especially because the `writeText.then` api is kinda abnormal, and naming the fns "handleClipboardWriteSuccess" and "...Error" help explain what's going on.